src/context/AccountsContext.tsx

```diff
@@ -1,84 +1,94 @@
-import {useApi} from 'src/context/ChainApiContext';
 
-export type Account = {
+export type InternalAccount = {
+  encoded: string;
   address: string;
-  name: string;
-  isFavorite: boolean;
-  isExternal: boolean;
-  network: SupportedNetworkType;
+  meta: {
+    name: string;
+    isFavorite: boolean;
+    network: SupportedNetworkType;
+  };
+  isExternal: false;
 };
 
-type Accounts = Record<string, Account>;
-
-const AccountsContext = createContext<Accounts>({});
-
-function AccountsProvider({children}: {children: React.ReactNode}) {
-  const {currentNetwork} = useContext(NetworkContext);
-  const [accounts, setAccounts] = useState<Accounts>({});
-  const {api} = useApi();
-
-  useEffect(() => {
-    let unsubscribe: VoidFn | undefined;
-
-    if (api) {
-      const subscription = keyring.accounts.subject.subscribe((accounts) => {
-        const keyringAccounts = Object.values(accounts).reduce((map, {json}) => {
-          const pair = tryGetPair(json.address);
-          if (json.meta.network === currentNetwork.key && pair) {
-            map = {
-              ...map,
-              [pair.address]: {
-                address: pair.address,
-                name: json.meta.name,
-                isFavorite: Boolean(json.meta.isFavorite),
-                isExternal: Boolean(json.meta.isExternal),
-                network: json.meta.network as SupportedNetworkType,
-              },
-            };
-          }
-          return map;
-        }, {});
+export type ExternalAccount = {
+  address: string;
+  meta: {
+    name: string;
+    isFavorite: boolean;
+    network: SupportedNetworkType;
+  };
+  isExternal: true;
+};
 
-        setAccounts(keyringAccounts);
-      });
+export type Account = InternalAccount | ExternalAccount;
 
-      unsubscribe = subscription.unsubscribe.bind(subscription);
-    }
+type Accounts = Record<string, Account>;
 
-    return () => {
-      unsubscribe && unsubscribe();
-    };
-  }, [currentNetwork, api]);
+type Context = {
+  accounts: Accounts;
+  addAccount: (account: Account) => void;
+  setAccountFavorite: (address: string, isFavorite: boolean) => void;
+  removeAccount: (address: string) => void;
+};
 
-  return <AccountsContext.Provider value={accounts}>{children}</AccountsContext.Provider>;
-}
+const AccountsContext = createContext<Context>({
+  accounts: {},
+  addAccount: () => {
+    return;
+  },
+  setAccountFavorite: () => {
+    return;
+  },
+  removeAccount: () => {
+    return;
+  },
+});
 
-function tryGetPair(address: string) {
-  try {
-    return keyring.getPair(address);
-  } catch (e) {
-    return undefined;
-  }
+function AccountsProvider({children}: {children: React.ReactNode}) {
+  const [accounts, setAccounts] = usePersistedState<Accounts>('accounts', {});
+
+  const value = {
+    accounts,
+    addAccount: (account: Account) => {
+      setAccounts({...accounts, [account.address]: account});
+    },
+    setAccountFavorite: (address: string, isFavorite: boolean) => {
```

Comment:
how about `toggleFavorite(address: string) => void` and change the value here instead of passing `isFavorite` from the callers

Reply:
done ✅